openassessment/xblock/grade_mixin.py

```diff
@@ -266,32 +252,189 @@ def _rubric_criteria_grade_context(self, peer_assessments, self_assessment):
+            example_based_assessment, self_assessment
+    ):
+        """
+        Returns an array of assessments with their associated grades.
+        """
+        def _get_assessment_part(title, part_criterion_name, assessment):
+            """
+            Returns the assessment part for the given criterion name.
+            """
+            if assessment:
+                for part in assessment['parts']:
+                    if part['criterion']['name'] == part_criterion_name:
+                        part['title'] = title
+                        return part
+            return None
+
+        criterion_name = criterion['name']
+        staff_assessment_part = _get_assessment_part(_('Staff Grade'), criterion_name, staff_assessment)
+        peer_assessment_part = {
+            'title': _('Peer Median Grade'),
+            'criterion': criterion,
+            'option': self._peer_median_option(submission_uuid, criterion),
+            'individual_assessments': [
+                _get_assessment_part(
+                    _('Peer {peer_index}').format(peer_index=index + 1),
+                    criterion_name,
+                    peer_assessment
+                )
+                for index, peer_assessment in enumerate(peer_assessments)
+                ],
+        } if len(peer_assessments) > 0 else None
+        example_based_assessment_part = _get_assessment_part(
+            _('Example-Based Grade'), criterion_name, example_based_assessment
+        )
+        self_assessment_part = _get_assessment_part(_('Your Self Assessment'), criterion_name, self_assessment)
+
+        # TODO: automate this...
+        assessments = []
+        if staff_assessment_part:
+            assessments.append(staff_assessment_part)
+        if peer_assessment_part:
+            assessments.append(peer_assessment_part)
+        if example_based_assessment_part:
+            assessments.append(example_based_assessment_part)
+        if self_assessment_part:
+            assessments.append(self_assessment_part)
+
+        # Include points only for the first assessment
+        if len(assessments) > 0:
+            option = assessments[0]['points'] = assessments[0]['option']
+            if option:
+                assessments[0]['points'] = option['points']
+
+        return assessments
+
+    def _peer_median_option(self, submission_uuid, criterion):
+        """
+        Returns the option for the median peer grade.
+
+        Args:
+            submission_uuid (str): The id for the submission.
+            criterion (dict): The criterion in question.
+
+        Returns:
+            The option for the median peer grade.
+
+        """
+        median_scores = peer_api.get_assessment_median_scores(submission_uuid)
+        median_score = median_scores.get(criterion['name'], None)
+        return next((option for option in criterion['options'] if option['points'] == median_score), None)
```

Comment:
What if the point value either doesn't match one of the textual representations, or matches more than one (like in the test course, where "Good" and "Excellent" each correspond to the same number of points)?